crates/soa/src/soa_slice.rs

```diff
@@ -29,39 +28,49 @@ impl<T, U> fmt::Debug for Slice<T, U> {
 
 // derive of copy and clone does not play well with PhantomData
 
-impl<T, U> Copy for Slice<T, U> {}
+impl<T> Copy for Slice<T> {}
 
-impl<T, U> Clone for Slice<T, U> {
+impl<T> Clone for Slice<T> {
     fn clone(&self) -> Self {
         *self
     }
 }
 
-impl<T, U> Default for Slice<T, U> {
+impl<T> Default for Slice<T> {
     fn default() -> Self {
         Self::empty()
     }
 }
 
-impl<Array, Elem> Slice<Array, Elem> {
-    pub fn empty() -> Self {
+impl<T> Slice<T> {
+    pub const fn empty() -> Self {
         Self {
             start: 0,
             length: 0,
-            _marker: Default::default(),
+            _marker: PhantomData,
         }
     }
 
-    pub fn get_slice<'a>(&self, slice: &'a [Elem]) -> &'a [Elem] {
-        &slice[self.indices()]
+    /// This is unsafe because it doesn't verify that the start index being returned is being used with the original
+    /// slice it was created with. Self::get_in is the safe alternative to this.
+    pub const fn start(self) -> u32 {
```

Comment:
If this is unsafe, can we name it in a way to tell consumers at the call site? Maybe `start_unchecked`?

Reply:
Oh that comment is obsolete - deleted!